helm-charts/splunk-otel-collector/templates/_helpers.tpl

```diff
@@ -90,7 +90,7 @@ Whether logs enabled for Splunk Platform.
 Whether metrics enabled for Splunk Platform.
 */}}
 {{- define "splunk-otel-collector.platformMetricsEnabled" -}}
-{{- and (eq (include "splunk-otel-collector.splunkPlatformEnabled" .) "true") .Values.splunkPlatform.metricsEnabled }}
+{{- and (eq (include "splunk-otel-collector.splunkPlatformEnabled" .) "true") .Values.splunkPlatform.metricsEnabled (not (eq .Values.splunkPlatform.metrics_index "")) }}
```

Comment:
Why metrics only? Shouldn't it be enforced for logs as well? I see `splunkPlatform.index` is optional. 

I believe this validation should better go into values.schema.json. Otherwise it can be confusing for a user when if they set `splunkPlatform.metricsEnabled=true` and it's silently ignored

Reply:
when index is not provided, there is still a default index that HEC token is associated with. That is typically event type index but that can be a metric type index too. so it makes sense to have a same behavior. okay i will implement this in `values.schema.json`